src/server10.py

```diff
@@ -18,123 +21,104 @@ def handle_connection(conx):
-    elif "cookie" in headers:
-        username = TOKENS.get(parse_cookies(headers["cookie"]).get("token"))
-
-    if method == "GET":
-        if url == "/":
-            return "200 OK", show_comments(username), resp_headers
-        elif url == "/login":
-            return "200 OK", login_form(), resp_headers
-        elif url == "/comment.js":
-            with open("comment9.js") as f:
-                return "200 OK", f.read(), resp_headers
-        elif url == "/comment.css":
-            with open("comment9.css") as f:
-                return "200 OK", f.read(), resp_headers
-    elif method == "POST":
-        if url == "/add":
-            params = form_decode(body)
-            return "200 OK", add_entry(params, username), resp_headers
-        else:
-            return "200 OK", show_comments(username), resp_headers
-
-    return "404 Not Found", not_found(url, method), resp_headers
-
-def login_form():
-    body = "<!doctype html>"
-    body += "<form action=/ method=post>"
-    body += "<p>Username: <input name=username></p>"
-    body += "<p>Password: <input name=password type=password></p>"
-    body += "<p><button>Log in</button></p>"
-    body += "</form>"
-    return body
+        return "200 OK", add_entry(session, params)
+    elif method == "POST" and url == "/login":
+        params = form_decode(body)
+        return do_login(session, params)
+    else:
+        return "404 Not Found", not_found(url, method)
 
-def html_escape(text):
-    return text.replace("&", "&amp;").replace("<", "&lt;")
+def form_decode(body):
+    params = {}
+    for field in body.split("&"):
+        name, value = field.split("=", 1)
+        params[name] = urllib.parse.unquote(value)
+    return params
 
-def show_comments(username):
+def show_comments(session):
     out = "<!doctype html>"
-    if username:
-        nonce = str(random.random())[2:]
-        NONCES[username] = nonce
+    if "user" in session:
         out += "<form action=add method=post>"
-        out +=   "<p><input name=nonce type=hidden value=" + nonce + "></p>"
         out +=   "<p><input name=guest></p>"
         out +=   "<p><button>Sign the book!</button></p>"
         out += "</form>"
     else:
-        out += "<p><a href=/login>Log in to add to the guest list</a></p>"
+        out += "<a href=/login>Sign in to write in the guest book</a>"
 
     for entry, who in ENTRIES:
-        out += "<p>" + html_escape(entry)
-        out += " <i>from " + who + "</i></p>"
+        out += "<p>" + entry + " <i>by " + who + "</i></p>"
+
+    out += "<link rel=stylesheet src=/comment.css>"
+    out += "<label></label>"
     out += "<script src=/comment.js></script>"
     return out
 
-def check_nonce(params, username):
-    if 'nonce' not in params: return False
-    if username not in NONCES: return False
-    return params['nonce'] == NONCES[username]
+def login_form(session):
+    body = "<!doctype html>"
+    body += "<form action=/login method=post>"
```

Comment:
Why should this be at "/"?  The login operation should end up at "/", so either that needs to be the POST destination or we need to support redirects.

Reply:
Yeah, you're right, I'll change this.